### PR TITLE
Updates from running scripts by different user for 0.24.2

### PR DESCRIPTION
**Pull request by jorisvandenbossche**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -64,7 +69,7 @@ conda-test:
 		-v ${CURDIR}/pandas:/pandas \
 		-v ${CURDIR}/recipe:/recipe \
 		pandas-build \
-		sh -c "conda build --numpy=1.12 /recipe --output-folder=/pandas/dist"
```
Comment (jreback) [inline Makefile]: maybe make these env variables

File: Makefile [hunk 2]
```diff
@@ -89,7 +94,7 @@ doc:
 upload-doc:
 	rsync -rv -e ssh pandas/doc/build/html/            pandas.pydata.org:/usr/share/nginx/pandas/pandas-docs/version/$(PANDAS_VERSION)/
 	rsync -rv -e ssh pandas/doc/build/latex/pandas.pdf pandas.pydata.org:/usr/share/nginx/pandas/pandas-docs/version/$(PANDAS_VERSION)/pandas.pdf
-	ssh pandas.pydata.org "cd /usr/share/nginx/pandas/pandas-docs && ln -sfn version/$(PANDAS_VERSION) stable && cd version && ln -sfn $(PANDAS_VERSION) $(PANDAS_VERSION:%.0=%)"
+	ssh pandas.pydata.org "cd /usr/share/nginx/pandas/pandas-docs && ln -sfn version/$(PANDAS_VERSION) stable && cd version && ln -sfn $(PANDAS_VERSION) $(PANDAS_VERSION:%.2=%)"
```
Comment (jorisvandenbossche) [inline Makefile]: This is not very robust. 
But didn't yet find a good way to remove the last two characters from a string

**Comment by TomAugspurger** [comment]
@jorisvandenbossche what doc build failure did you see? Something about tornado.web?

My doc env had nbcovert 5.3.1 and tornado 6.0.1. Tornado 6.x requires nbconvert 5.4.1+

I believe Anaconda patched the metadata to reflect this constraint this morning. In the meantime, I'm manually updating to 5.4.1 in the doc build env.

**Comment by TomAugspurger** [comment]
One more TODO that I'll just lodge here, we had some difficulties with uploaded the HTML / pdf docs to the webserver, since I Joris couldn't modify files owned by my user, and I couldn't modify files owned by Joris.

It'd be best to have a `pandas` user or group, and have the files owned by them if possible.

**Comment by jorisvandenbossche** [comment]
> @jorisvandenbossche what doc build failure did you see? Something about tornado.web?

It might actually be that the pdf generation worked, but it only messages a failure. When rerunning it again now, the output shows:

```
...
Output written on pandas.pdf (2999 pages, 8823241 bytes).
Transcript written on pandas.log.
Traceback (most recent call last):
  File "./make.py", line 339, in <module>
    sys.exit(main())
  File "./make.py", line 335, in main
    return getattr(builder, args.command)()
  File "./make.py", line 253, in latex_forced
    return self.latex(force=True)
  File "./make.py", line 242, in latex
    'pandas.tex')
  File "./make.py", line 104, in _run_os
    subprocess.check_call(args, stdout=sys.stdout, stderr=sys.stderr)
  File "/opt/conda/envs/pandas/lib/python3.6/subprocess.py", line 311, in check_call
    raise CalledProcessError(retcode, cmd)
subprocess.CalledProcessError: Command '('pdflatex', '-interaction=nonstopmode', 'pandas.tex')' returned non-zero exit status 1.
Makefile:86: recipe for target 'doc' failed
make: *** [doc] Error 1
```

So actually the pdf seems to be generated, but the process returns a non-zero exit code.

**Comment by jorisvandenbossche** [comment]
Moved a few obvious things (pytest-mock and passing of username to wheels script) to separate PRs + merged those.

File: Makefile [hunk 3]
```diff
@@ -1,7 +1,13 @@
-TAG ?= v0.24.1
+# TO EDIT
+TAG ?= v0.24.2
+GH_USERNAME ?= jorisvandenbossche
+
 PANDAS_VERSION=$(TAG:v%=%)
+PANDAS_BASE_VERSION=0.24
```
Comment (jorisvandenbossche) [inline Makefile]: I didn't find a way to automatically convert something like "0.24.2" into "0.24" with make, so for now just hardcoded it here (so it should be in the section "to edit"). 
Anybody an idea how to do this automatically?

Comment (TomAugspurger) [reply Makefile]: Pushed an awk version.

**Pull request merged**